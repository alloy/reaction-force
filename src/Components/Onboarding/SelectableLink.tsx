import React from "react"
import styled, { StyledFunction } from "styled-components"

import * as fonts from "../../Assets/Fonts"
import Icon from "../Icon"

interface SelectableLinkProps {
  href?: string
  text?: string
  onSelect: (selected: boolean) => void
}

interface LinkState {
  selected: boolean
}

interface IconContainerProps extends React.HTMLProps<HTMLIFrameElement> {
  isSelected: boolean
}

const iconContainer: StyledFunction<IconContainerProps> = styled.div

const IconContainer = iconContainer`
  width: 18px;
  height: 18px;
  background-color: black;
  border-radius: 50%;
  float: right;
  margin-right: 15px;
  display: ${props => (props.isSelected ? "inline-flex;" : "none;")}
  justify-content: ${props => (props.isSelected ? "center;" : "none;")}
`

const Link = styled.a`
  display: block;
  font-size: 14px;
  color: black;
  text-decoration: none;
  text-transform: uppercase;
  font-family: ${fonts.primary.fontFamily};
  padding: 30px 0 30px 15px;
  border-top: 1px solid #e5e5e5;
  &:hover {
    background-color: #f8f8f8;
  }
  &:hover ${IconContainer} {
    display: inline-flex;
    justify-content: center;
  }
`

class SelectableLink extends React.Component<SelectableLinkProps, LinkState> {
  constructor(props) {
    super(props)
    this.state = {
      selected: false,
    }
    this.onSelect = this.onSelect.bind(this)
  }

  onSelect() {
    const state = !this.state.selected
    this.setState({ selected: state })
    if (this.props.onSelect) {
      this.props.onSelect(state)
    }
  }

  render() {
    return (
      <div>
        <Link href={this.props.href} onClick={() => this.onSelect()}>
          {this.props.text}

          <IconContainer isSelected={this.state.selected}>
            <Icon name="follow-circle.is-following" color="white" fontSize="39px" style={{ alignSelf: "center" }} />
          </IconContainer>
        </Link>
      </div>
    )
  }
}

export default SelectableLink
