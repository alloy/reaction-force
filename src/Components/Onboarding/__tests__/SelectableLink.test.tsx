import "jest-styled-components"
import React from "react"
import renderer from "react-test-renderer"
import SelectableLink from "../SelectableLink"

describe("SelectableLink", () => {
  describe("snapshot", () => {
    it("renders a selectable link", () => {
      const component = renderer
        .create(
          <SelectableLink
            href="/artist/andy-warhol"
            text="Andy"
            onSelect={jest.fn()}
          />
        )
        .toJSON()
      expect(component).toMatchSnapshot()
    })
  })
})
