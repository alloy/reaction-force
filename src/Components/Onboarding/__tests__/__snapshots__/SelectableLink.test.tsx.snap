// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectableLink snapshot renders a selectable link 1`] = `
.c3 {
  font-family: artsy-icons;
  color: white;
  font-size: 39px;
  margin: 0 5px;
  display: inline-block;
  -webkit-letter-spacing: 0px;
  -moz-letter-spacing: 0px;
  -ms-letter-spacing: 0px;
  letter-spacing: 0px;
}

.c2 {
  width: 18px;
  height: 18px;
  background-color: black;
  border-radius: 50%;
  float: right;
  margin-right: 15px;
  display: none;
  -webkit-box-pack: none;
  -webkit-justify-content: none;
  -ms-flex-pack: none;
  justify-content: none;
}

.c0 {
  display: block;
  font-size: 14px;
  color: black;
  text-decoration: none;
  font-family: 'ITC Avant Garde Gothic W04', 'AvantGardeGothicITCW01D 731075', 'AvantGardeGothicITCW01Dm', 'Helvetica', 'sans-serif';
  -webkit-font-smoothing: antialiased;
  text-transform: uppercase;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  padding: 30px 0 30px 15px;
  border-top: 1px solid #e5e5e5;
}

.c0:hover {
  background-color: #f8f8f8;
  cursor: pointer;
}

.c0:hover .c1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

@media (max-width:48em) {
  .c0 {
    padding: 15px 0 15px 15px;
  }
}

<div>
  <a
    className="c0"
    href="/artist/andy-warhol"
    onClick={[Function]}
  >
    Andy
    <div
      className="c1 c2"
    >
      <div
        className="c3"
        onClick={undefined}
        style={
          Object {
            "alignSelf": "center",
          }
        }
      >
        î˜¨
      </div>
    </div>
  </a>
</div>
`;
